cli/azd/internal/cmd/add/add.go

```diff
@@ -0,0 +1,392 @@
+	if !a.alphaManager.IsEnabled(composeFeature) {
+		return nil, fmt.Errorf(
+			"compose is currently under alpha support and must be explicitly enabled."+
+				" Run `%s` to enable this feature", alpha.GetEnableCommand(composeFeature),
+		)
+	}
+
+	prjConfig, err := project.Load(ctx, a.azdCtx.ProjectPath())
+	if err != nil {
+		return nil, err
+	}
+
+	selectMenu := a.selectMenu()
+	slices.SortFunc(selectMenu, func(a, b Menu) int {
+		return strings.Compare(a.Label, b.Label)
+	})
+
+	selections := make([]string, 0, len(selectMenu))
+	for _, menu := range selectMenu {
+		selections = append(selections, menu.Label)
+	}
+	idx, err := a.console.Select(ctx, input.ConsoleOptions{
+		Message: "What would you like to add?",
+		Options: selections,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	selected := selectMenu[idx]
+
+	resourceToAdd := &project.ResourceConfig{}
+	var serviceToAdd *project.ServiceConfig
+
+	ctxPrompt := promptCtx{Context: ctx, prj: prjConfig}
+	if strings.EqualFold(selected.Namespace, "host") {
+		svc, r, err := a.configureHost(a.console, ctxPrompt)
+		if err != nil {
+			return nil, err
+		}
+
+		resourceToAdd = r
+		serviceToAdd = svc
+	} else {
+		r, err := selected.SelectResource(a.console, ctxPrompt)
+		if err != nil {
+			return nil, err
+		}
+
+		resourceToAdd = r
+	}
+
+	r, err := fillRequired(resourceToAdd, a.console, ctxPrompt)
+	if err != nil {
+		return nil, err
+	}
+
+	resourceToAdd = r
+
+	usedBy := []string{}
+	if strings.HasPrefix(string(resourceToAdd.Type), "host.") {
+		r, err := fillUsesForHost(resourceToAdd, a.console, ctxPrompt)
+		if err != nil {
+			return nil, err
+		}
+
+		resourceToAdd = r
+	} else {
+		uses, err := promptUses(resourceToAdd, a.console, ctxPrompt)
+		if err != nil {
+			return nil, err
+		}
+
+		usedBy = uses
+	}
+
+	file, err := os.OpenFile(a.azdCtx.ProjectPath(), os.O_RDWR, osutil.PermissionFile)
+	if err != nil {
+		return nil, fmt.Errorf("reading project file: %w", err)
+	}
+	defer file.Close()
+
+	decoder := yaml.NewDecoder(file)
+	decoder.SetScanBlockScalarAsLiteral(true)
+
+	var doc yaml.Node
+	err = decoder.Decode(&doc)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode: %w", err)
+	}
+
+	if serviceToAdd != nil {
+		serviceNode, err := yamlnode.Encode(serviceToAdd)
+		if err != nil {
+			panic(fmt.Sprintf("encoding yaml node: %v", err))
+		}
+
+		err = yamlnode.Set(&doc, fmt.Sprintf("services?.%s", serviceToAdd.Name), serviceNode)
+		if err != nil {
+			return nil, fmt.Errorf("finding services: %w", err)
+		}
+	}
+
+	resourceNode, err := yamlnode.Encode(resourceToAdd)
+	if err != nil {
+		panic(fmt.Sprintf("encoding yaml node: %v", err))
+	}
+
+	err = yamlnode.Set(&doc, fmt.Sprintf("resources?.%s", resourceToAdd.Name), resourceNode)
+	if err != nil {
+		return nil, fmt.Errorf("setting resource: %w", err)
+	}
+
+	for _, svc := range usedBy {
+		err = yamlnode.Append(&doc, fmt.Sprintf("resources.%s.uses[]?", svc), &yaml.Node{
```

Comment:
verify `uses` beforehand for duplicates somehow

Reply:
no duplicates possible; but I did end up exposing host bi-directional uses and hardened it to avoid cycles there